### PR TITLE
Migrate more tests from jpa/test to orm/test/jpa

### PR DESCRIPTION
**Pull request by jrenaat**
Signed-off-by: Jan Schatteman <jschatte@redhat.com>

### CODE REVIEW

File: hibernate-core/src/test/java/org/hibernate/orm/test/jpa/lob/BlobTest.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+import org.hibernate.testing.orm.junit.Jpa;
+
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+/**
+ * @author Emmanuel Bernard
+ */
+@RequiresDialectFeature( DialectChecks.SupportsExpectedLobUsagePattern.class )
```
Comment (dreab8) [inline hibernate-core/src/test/java/org/hibernate/orm/test/jpa/lob/BlobTest.java:29]: It should be `@RequiresDialectFeature( feature = DialectFeatureChecks.SupportsExpectedLobUsagePattern.class )
`

File: hibernate-core/src/test/java/org/hibernate/orm/test/jpa/mapping/UnidirectionalOneToManyIndexColumnTest.java [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+		annotatedClasses = {
+				UnidirectionalOneToManyIndexColumnTest.Parent.class,
+				UnidirectionalOneToManyIndexColumnTest.Child.class
+		},
+		integrationSettings = {
+				@Setting(name = AvailableSettings.USE_NEW_ID_GENERATOR_MAPPINGS, value = "true"),
+				@Setting(name = AvailableSettings.HBM2DDL_AUTO, value = "create-drop")
+		}
+)
+public class UnidirectionalOneToManyIndexColumnTest {
```
Comment (beikov) [inline hibernate-core/src/test/java/org/hibernate/orm/test/jpa/mapping/UnidirectionalOneToManyIndexColumnTest.java:46]: For some reason, this tests seems to fail on Derby.

**Review by beikov — CHANGES_REQUESTED** [review]
Looks good, except for the test failure on Derby and what Andrea suggested.

File: hibernate-core/src/test/java/org/hibernate/orm/test/jpa/lob/BlobTest.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+			em.close();
+		}
+		catch (Exception e) {
+			if ( em.getTransaction().isActive() ) {
+				em.getTransaction().rollback();
+			}
+			throw e;
+		}
+	}
+}
```
Comment (dreab8) [inline hibernate-core/src/test/java/org/hibernate/orm/test/jpa/lob/BlobTest.java:68]: What something like 
```
Long readerId = scope.fromTransaction(
				entityManager -> {
					try {
						Map<String,String> image = new HashMap<>();
						image.put( "meta", "metadata" );
						image.put( "data", "imagedata" );
						ImageReader reader = new ImageReader();
						ByteArrayOutputStream baos = new ByteArrayOutputStream();
						ObjectOutputStream oos = new ObjectOutputStream( baos );
						oos.writeObject( image );
						reader.setImage( entityManager.unwrap( Session.class ).getLobHelper().createBlob( baos.toByteArray() ) );
						entityManager.persist( reader );
						return reader.getId();
					}
					catch (IOException e) {
						throw new RuntimeException(e);
					}
				}
		);

		scope.inTransaction(
				entityManager -> {
					try {
						ImageReader reader = entityManager.find( ImageReader.class, readerId );
						ObjectInputStream ois = new ObjectInputStream( reader.getImage().getBinaryStream() );
						Map<String, String> image = (HashMap<String, String>) ois.readObject();
						assertTrue( image.containsKey( "meta" ) );
					}
					catch (Exception e) {
						throw new RuntimeException( e );
					}
				}
		);
```

**Pull request merged**